### PR TITLE
more meaningful units than Hz

### PR DESCRIPTION
**Pull request by honnix**

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 1]
```diff
@@ -60,43 +59,43 @@
...
 
   private static final MetricId TERMINATION_LOG_INVALID = BASE
       .tagged("what", "termination-log-invalid")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "count");
 
   private static final MetricId EXIT_CODE_MISMATCH = BASE
       .tagged("what", "exit-code-mismatch")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "count");
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: As this is used for a meter, `count` would be emitted as `count/s` which doesn't seem to make much sense. How about e.g. `mismatch` instead? We could also omit the unit and fall back to the default of `n` which is emitted as `n/s`.

Similar for the rest of the units.

https://github.com/spotify/semantic-metrics#meter

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 2]
```diff
@@ -60,43 +59,43 @@
...
 
   private static final MetricId TERMINATION_LOG_MISSING = BASE
       .tagged("what", "termination-log-missing")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "count");
 
   private static final MetricId TERMINATION_LOG_INVALID = BASE
       .tagged("what", "termination-log-invalid")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "count");
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: Omit unit and use default?

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 3]
```diff
@@ -60,43 +59,43 @@
...
 
   private static final MetricId NATURAL_TRIGGER_RATE = BASE
       .tagged("what", "natural-trigger-rate")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "triggering");
 
   private static final MetricId TERMINATION_LOG_MISSING = BASE
       .tagged("what", "termination-log-missing")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "count");
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: Omit unit and use default?

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 4]
```diff
@@ -60,43 +59,43 @@
...
 
   private static final MetricId PULL_IMAGE_ERROR_RATE = BASE
       .tagged("what", "pull-image-error-rate")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "error");
 
   private static final MetricId NATURAL_TRIGGER_RATE = BASE
       .tagged("what", "natural-trigger-rate")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "triggering");
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: `s/triggering/trigger` ?

`triggering/s` vs `trigger/s`

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java [hunk 5]
```diff
@@ -60,43 +59,43 @@
...
 
   private static final MetricId EXIT_CODE_MISMATCH = BASE
       .tagged("what", "exit-code-mismatch")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "count");
 
   private static final MetricId SUBMISSION_RATE_LIMIT = BASE
       .tagged("what", "submission-rate-limit")
-      .tagged("unit", UNIT_FREQUENCY);
+      .tagged("unit", "submission");
```
Comment (danielnorberg) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: `submission` -> `submission/s` ?

https://github.com/spotify/semantic-metrics#gauge

**Comment by danielnorberg** [comment]
There's some trickiness wrt units of meters and gauges etc to consider here.

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java (hunk 1 above)
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: Using default looks better.

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java (hunk 5 above)
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: Yes, sure. Didn't realize it's not a meter.

File: styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java (hunk 4 above)
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/monitoring/MetricsStats.java]: Will fix.

**Comment by danielnorberg** [comment]
👍  nice!

**Pull request merged**